src/ethereum/forks/amsterdam/vm/eoa_delegation.py

```diff
@@ -155,51 +149,16 @@ def calculate_delegation_cost(
     track_address(evm.state_changes, address)
 
     if not is_valid_delegation(code):
-        return False, address, None, Uint(0)
+        return False, address, Uint(0)
 
     delegated_address = Address(code[EOA_DELEGATION_MARKER_LENGTH:])
 
-    # Calculate gas cost for delegation target access
     if delegated_address in evm.accessed_addresses:
         delegation_gas_cost = GAS_WARM_ACCESS
     else:
         delegation_gas_cost = GAS_COLD_ACCOUNT_ACCESS
 
-    return True, address, delegated_address, delegation_gas_cost
-
-
-def read_delegation_target(evm: Evm, delegated_address: Address) -> Bytes:
```

Comment:
This is unnecessary. By the time we get code for the `code_address`, we already have either the account as the `code_address` or the delegated address, so we can simply call `get_account(state, code_address).code`, essentially.